### DISCUSSION TITLE
Exporting code error

### DISCUSSION THREAD

**Discussion by dariomarino01** [HALRobotics/Beta#125] [Q&A] [answered]
Hi all!,
I've got an error when trying to export the code for the robot (see attached image).
The simulation works smoothly without any errors.
What could it be the problem?
I also tried to restart Rhino and Hal, it works sometimes but not his one :)
I am new to Hal, forgive me if I made any silly mistake or posted in the wrong place. I'm learning...
Thanks for your time.

![image](https://user-images.githubusercontent.com/105284792/167613356-24a7a016-29cd-452f-ab89-bcc9d1ea7bb2.png)

**Comment by sebandraos** [comment, marked as answer]
Hi @dariomarino01 , This definitely looks like an issue. Can you share your Rhino and Grasshopper files (or preferably a simplified version) so we can look at why this might be occurring? Thanks

**Reply by dariomarino01** [reply]
Hi @sebandraos ,
here are the files.
thank you!

[Cambridge_Cell.zip](https://github.com/HALRobotics/Beta/files/8662069/Cambridge_Cell.zip)

**Reply by sebandraos** [reply]
Hi @dariomarino01 ,
There are two things missing here:
1. The mapping of the turntable joint has moved in our components and it doesn't look like your file adheres to this change. In the image below the "Mapping" should be passed into the Assemble component.
![image](https://user-images.githubusercontent.com/6084391/167835127-61a926bb-a495-4acb-930b-943ff76c4480.png)

**There is a little issue with this in the current release so please look out for an update later today which will fix that.**

2. The turntable procedure needs to be mapped as a child procedure of the robot's procedure. You can read more about this [here](https://docs.hal-robotics.com/Grasshopper/6-Control/Contents.html#62-export-a-procedure). Dragging and dropping the Turntable Procedure onto the Robot procedure in the Controller Configurator will sort that out.

**Reply by dariomarino01** [reply]
Thank you,
I did try to add the mapping before and it was actually giving problems.
Thanks a lot for looking into that.
